lib/nostrum/api.ex

```diff
@@ -1229,89 +1229,163 @@ defmodule Nostrum.Api do
   @doc """
   Gets a guild member.
 
-  Member to get is specified by `guild_id` and `user_id`.
+  If successful, returns `{:ok, member}`. Otherwise, returns a `t:Nostrum.Api.error/0`.
+
+  ## Examples
+
+  ```Elixir
+  Nostrum.Api.get_guild_member(4019283754613, 184937267485)
+  ```
   """
-  @spec get_member(integer, integer) :: error | {:ok, Nostrum.Struct.Guild.Member.t()}
-  def get_member(guild_id, user_id) do
-    case request(:get, Constants.guild_member(guild_id, user_id)) do
-      {:ok, body} ->
-        {:ok, Poison.decode!(body)}
+  @spec get_guild_member(Guild.id(), User.id()) :: error | {:ok, Member.t()}
+  def get_guild_member(guild_id, user_id) when is_snowflake(guild_id) and is_snowflake(user_id) do
+    request(:get, Constants.guild_member(guild_id, user_id))
+    |> handle_request_with_decode({:struct, Member})
+  end
 
-      other ->
-        other
-    end
+  @doc """
+  Same as `get_guild_member/2`, but raises `Nostrum.Error.ApiError` in case of failure.
+  """
+  @spec get_guild_member!(Guild.id(), User.id()) :: no_return | Member.t()
+  def get_guild_member!(guild_id, user_id) do
+    get_guild_member(guild_id, user_id)
+    |> bangify
   end
 
   @doc """
-  Gets a list of guild members.
+  Gets a list of a guild's members.
 
-  ## Parameters
-  `guild_id` - Guild to get members.
-  `options` - Map with the following *optional* keys:
-    - `limit` - Max number of members to return (1-1000)
-    - `after` - Highest user id of the previous page.
-  """
-  @spec get_guild_members(Guild.id(), %{
-          limit: 1..1000,
-          after: integer
-        }) :: error | {:ok, [Nostrum.Struct.Guild.Member.t()]}
-  def get_guild_members(guild_id, options) do
+  If successful, returns `{:ok, members}`. Otherwise, returns a `t:Nostrum.Api.error/0`.
+
+  ## Options
+
+    * `:limit` (integer) - max number of members to return (1-1000) (default: 1)
+    * `:after` (`t:Nostrum.Struct.Snowflake.t/0`) - the highest user id in the previous page (default: 0)
```

Comment:
I saw you switched this from `User.id()` to `Snowflake.t`, what was the reasoning behind this?

Reply:
I was experimenting with the types a little bit to see which one would be more informative to our users.  Looks like I accidentally sent this change with my self code-review. I've fixed it with the recent commit.